src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -53,11 +53,29 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
         globalError(s"Failed to parse `-Wconf` configuration: ${settings.Wconf.value}\n${msgs.mkString("\n")}$multiHelp")
         WConf(Nil)
       case Right(conf) =>
-        // configure cat=scala3-migration if it isn't yet
-        val Migration = MessageFilter.Category(WarningCategory.Scala3Migration)
-        val boost = (settings.isScala3: @nowarn) && !conf.filters.exists(_._1.exists(_ == Migration))
-        if (boost) conf.copy(filters = conf.filters :+ (Migration :: Nil, Action.Error))
-        else conf
+        // default is "cat=deprecation:ws,cat=feature:ws,cat=optimizer:ws"
+        // under -deprecation, "cat=deprecation:w", but under -deprecation:false or -nowarn, "cat=deprecation:s"
+        // similarly for -feature, -Wopt (?)
+        val needsDefaultAdjustment = settings.deprecation.isSetByUser
+        val adjusted =
+          if (needsDefaultAdjustment || settings.nowarn.value) {
+            val (user, defaults) = conf.filters.splitAt(conf.filters.length - settings.WconfDefault.length)
+            val Deprecation = MessageFilter.Category(WarningCategory.Deprecation)
+            val action = if (settings.deprecation.value) Action.Warning else Action.Silent
```

Comment:
Worth following up whether `-deprecation -nowarn` does the obvious thing (whatever that is). With a custom reporter, there could be a difference between "what is displayed" and "warnings that reach the reporter"?